File: debop4s-core/src/main/scala/kr/debop4s/core/BinaryStringFormat.scala
package kr.debop4s.core

/**
 * kr.debop4s.core.BinaryStringFormat
 * @author 배성혁 sunghyouk.bae@gmail.com
 * @since  2013. 12. 9. 오후 10:20
 */
object BinaryStringFormat extends Enumeration {

    type BinaryStringFormat = Value

    val Base64 = Value("Base64")

    val HexDecimal = Value("HexDecimal")

}

File: debop4s-core/src/main/scala/debop4s/core/BinaryStringFormat.java
package debop4s.core;

/**
 * Byte 배열의 정보를 문자열로 표현할 때 사용할 형식 (Base64 또는 HexDecimal 방식)
 *
 * @author 배성혁 sunghyouk.bae@gmail.com
 * @since 13. 6. 28. 오후 5:08
 */
public enum BinaryStringFormat {

    /** Base64 인코딩 방식으로 문자열로 표현 */
    Base64("Base64"),

    /** 16진수 방식으로 문자열로 표현 */
    HexDecimal("HexDecimal");

    private final String value;

    public String getValue() {
        return value;
    }

    BinaryStringFormat(final String stringFormat) {
        this.value = stringFormat;
    }
}

